### PR TITLE
Go Select Group By

### PR DESCRIPTION
**Pull request by StevenUlmer**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md
~~Tests for the changes have been added (for bug fixes / features)~~
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Issue Number: https://github.com/mobi/goponents/issues/374

## What is the new behavior?
Select items can be grouped by a property that exists on each item.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-select/go-select.component.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ export class GoSelectComponent implements OnInit {
   @Input() bindLabel: string;
   @Input() bindValue: string;
   @Input() control: FormControl;
+  /**
+   * A property on each item to group by
+   */
```
Comment (deotte) [inline projects/go-lib/src/lib/components/go-select/go-select.component.ts]: Did you want to keep this?

File: projects/go-style-guide/src/app/features/ui-kit/components/form-docs/components/select-docs/select-docs.component.ts [hunk 2]
```diff
@@ -188,6 +197,27 @@ export class SelectDocsComponent implements OnInit {
   );
   `;
 
+  select12Code: string = `
+  <go-select
+    bindLabel="name"
+    bindValue="value"
+    [control]="select"
+    groupBy="manufacturer"
```
Comment (deotte) [inline projects/go-style-guide/src/app/features/ui-kit/components/form-docs/components/select-docs/select-docs.component.ts]: Assuming this has to match the *exact* property on your objects, right? Just asking for curiosity's sake.

File: projects/go-lib/src/lib/components/go-select/go-select.component.ts (hunk 1 above)
Comment (StevenUlmer) [reply projects/go-lib/src/lib/components/go-select/go-select.component.ts]: Yup this adds js docs for this

File: projects/go-style-guide/src/app/features/ui-kit/components/form-docs/components/select-docs/select-docs.component.ts (hunk 2 above)
Comment (StevenUlmer) [reply projects/go-style-guide/src/app/features/ui-kit/components/form-docs/components/select-docs/select-docs.component.ts]: Yup it has to match exactly

**Pull request merged**